### PR TITLE
fix(speech): enable location specific connections

### PR DESCRIPTION
**Pull request by scotthart**
part of the work for #13729 

In Speech V2 locations (aka regions) can be specified for recognizers, but the connection must also be configured for that same location. While explicitly setting `Options` can achieve this result, adding a location aware `MakeSpeechConnection` overload streamlines this process. When the location is provided to `MakeSpeechConnection`, the SDK handles setting the appropriate `Options` on behalf of the user.

e.g.
```
namespace speech = ::google::cloud::speech_v2;
auto client = speech::SpeechClient(speech::MakeSpeechConnection("asia-southeast1"));
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757)

### CODE REVIEW

File: google/cloud/speech/doc/environment-variables.dox [hunk 1]
```diff
@@ -14,6 +14,10 @@ environment variables are convenient when troubleshooting problems.
   `EndpointOption` (which defaults to "speech.googleapis.com")
   used by `MakeAdaptationConnection()`.
 
+- `GOOGLE_CLOUD_CPP_SPEECH_ENDPOINT=...` overrides the
+  `EndpointOption` (which defaults to "<location>-speech.googleapis.com")
+  used by `MakeSpeechConnection()`.
+
 - `GOOGLE_CLOUD_CPP_SPEECH_ENDPOINT=...` overrides the
   `EndpointOption` (which defaults to "speech.googleapis.com")
   used by `MakeSpeechConnection()`.
```
Comment (devbww) [inline google/cloud/speech/doc/environment-variables.dox:23]: Should both of these exist?

File: google/cloud/speech/README.md [hunk 2]
```diff
@@ -23,28 +23,49 @@ top-level [README](/README.md#building-and-installing).
...
               << kDefaultUri << "\n";
     return 1;
   }
-  auto uri = std::string{argc == 2 ? argv[1] : kDefaultUri};
-
-  namespace speech = ::google::cloud::speech_v1;
-  auto client = speech::SpeechClient(speech::MakeSpeechConnection());
+  std::string const project = argv[1];
+  std::string const location = argv[2];
+  std::cout << "project=" << project << "; location=" << location << "\n";
```
Comment (devbww) [inline google/cloud/speech/README.md:48]: Debugging remnant?

File: google/cloud/speech/quickstart/quickstart.cc [hunk 3]
```diff
@@ -13,28 +13,48 @@
...
+  google::cloud::speech::v2::RecognizeRequest request;
+  ConfigureRecognizer(request);
+  request.set_uri(uri);
+
+  if (location == "global") {
+    connection = speech::MakeSpeechConnection();
+    request.set_recognizer("projects/" + project +
+                           "/locations/global/recognizers/_");
+  } else {
+    connection = speech::MakeSpeechConnection(location);
```
Comment (devbww) [inline google/cloud/speech/quickstart/quickstart.cc:51]: The quickstart is built using the latest release, so probably this needs to be staged.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757)* status: 0 of 10 files reviewed, 3 unresolved discussions (waiting on @devbww)

___
*[`google/cloud/speech/doc/environment-variables.dox` line 23 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757#gh-1518408361:-NshycGZBGgfKMSjbAHB:b-r1h7sa) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/51375d3dade94a6286a5bbf102ecb40815009bae/google/cloud/speech/doc/environment-variables.dox#L23)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Should both of these exist?
</blockquote></details>

Apparently it's generated as a consequence of using `LOCATION_DEPENDENT_COMPAT`. We may want to revisit that in the future.
___
*[`google/cloud/speech/README.md` line 48 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757#gh-1518409137:-NshyaaC4PUkjDsGQ7Tv:bss4a2b) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c7443290a58246722731f1885bdb9adcf49293c3/google/cloud/speech/README.md#L48)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Debugging remnant?
</blockquote></details>

Removed.
___
*[`google/cloud/speech/quickstart/quickstart.cc` line 51 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757#gh-1518410610:-NshyVw2EeiDGJ2Vm_p4:b-36h53r) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c7443290a58246722731f1885bdb9adcf49293c3/google/cloud/speech/quickstart/quickstart.cc#L51)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

The quickstart is built using the latest release, so probably this needs to be staged.
</blockquote></details>

Unfortunately, it does need to be staged.

**Review by devbww — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757)* status: 0 of 10 files reviewed, all discussions resolved

___
*[`google/cloud/speech/doc/environment-variables.dox` line 23 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757#gh-1518408361:-NsiC6qm302IxQuTKgpG:b2ngt3r) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/51375d3dade94a6286a5bbf102ecb40815009bae/google/cloud/speech/doc/environment-variables.dox#L23)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

Apparently it's generated as a consequence of using `LOCATION_DEPENDENT_COMPAT`. We may want to revisit that in the future.
</blockquote></details>

+ 1
___
*[`google/cloud/speech/quickstart/quickstart.cc` line 51 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13757#gh-1518410610:-NsiCcTjCoKxQGvHfrZS:bb6sh2i) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c7443290a58246722731f1885bdb9adcf49293c3/google/cloud/speech/quickstart/quickstart.cc#L51)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

Unfortunately, it does need to be staged.
</blockquote></details>

Ack.

**Pull request merged**